### PR TITLE
Migrations-1015 - Add buildscript for replayer

### PR DESCRIPTION
**Pull request by okhasawn**
### Description
This adds a gradle build script file to make it possible to repeatedly build the Traffic Replayer, compiling all java files, run unit tests, checkstyle and spotbugs on the code, all with one command.

You can either run `gradle build` within the traffic replayer directory, or to do this with a gradle wrapper, then you must run `gradle wrapper` then run `./gradlew build`

### Issues Resolved
[Migrations-1015](https://opensearch.atlassian.net/browse/MIGRATIONS-1015)

### Check List
- [ ] New functionality includes testing
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by gregschohn** [comment]
Can you look up what the conventions are for other opensearch gradle projects.  What are some of the target names, are there any other specific tools that are commonly used (like checkstyle)?  You can create more tasks to add them, but it's good to maintain consistency for future developers.

**Comment by okhasawn** [comment]
> Can you look up what the conventions are for other opensearch gradle projects. What are some of the target names, are there any other specific tools that are commonly used (like checkstyle)? You can create more tasks to add them, but it's good to maintain consistency for future developers.

opensearch-project/opensearch-java repo actually has checkstyle as a [github action](https://github.com/opensearch-project/opensearch-java/actions/workflows/checkstyle.yml), while the opensearch-project/opensearch [stopped using checkstyle](https://github.com/opensearch-project/OpenSearch/pull/1370) and started using spotless

**Review by gregschohn — COMMENTED** [review]
I'm not familiar w/ a lot of the specific details that need to go into the build.gradle file, but I've left some comments.  I'd like to make sure that this works well with IDE integration and the command line.  Please add some notes to the PR with your findings.

File: TrafficReplayer/build.gradle [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    main {
+        java {
+            srcDirs = ['src/org/opensearch/migrations/replay','src/org/opensearch/migrations/replay/netty']
+        }
+    }
+
+    unitTests {
+        java {
+            compileClasspath += main.output + test.output
+            runtimeClasspath += main.output + test.output
```
[the remark this answers is not in the document]
Comment (okhasawn) [reply TrafficReplayer/build.gradle:61]: compileClasspath defaults to the compile<SourceSet> configuration and runtimeClasspath defaults to the sourceSet's output + runtime<SourceSet> configuration

File: TrafficReplayer/build.gradle [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    implementation group: 'com.beust', name: 'jcommander', version: '1.82'
+    implementation group: 'org.apache.logging.log4j', name: 'log4j-api', version: '2.20.0'
+    annotationProcessor 'org.projectlombok:lombok:1.18.10'
+
+}
+
+sourceSets {
+    main {
+        java {
+            srcDirs = ['src/org/opensearch/migrations/replay','src/org/opensearch/migrations/replay/netty']
```
[the remark this answers is not in the document]
Comment (okhasawn) [reply TrafficReplayer/build.gradle:54]: Thanks for letting me know and pointing this out. It does still build and load well into IntelliJ, I'll update this and "tst" too!

**Comment by lewijacn** [comment]
Couple things then I'm fine to approve:

I don't want to mess with the gradle configuration too much while this is still a sub directory of this repo as its hard to tell what is an actual problem or what is Intelij having trouble reconciling. But we should definitely make sure this works well when we move it out and is properly picked up by Intelij.

I was able to build successfully after fixing a spot bugs issue in line 282 of TrafficReplayer.java. Not sure if you've looked into this but if @gregschohn does not have concern would add it in
Bug: Found a call to a method which will perform a byte to String (or String to byte) conversion, and will assume that the default platform encoding is suitable. This will cause the application behaviour to vary between platforms. Use an alternative API and specify a charset name or Charset object explicitly
Original:
`try (InputStreamReader r = new InputStreamReader(fis)) {`
Change:
`try (InputStreamReader r = new InputStreamReader(fis, StandardCharsets.UTF_8)) {`

**Review by lewijacn — APPROVED** [review]
LGTM

**Review by gregschohn — APPROVED** [review]
You can also delete the IntelliJ project file (TrafficReplayer.iml)
I'm sure that I will soon regret not looking at the checkstyle.xml file carefully :)

**Comment by okhasawn** [comment]
> You can also delete the IntelliJ project file (TrafficReplayer.iml) I'm sure that I will soon regret not looking at the checkstyle.xml file carefully :)

@gregschohn I'll remove that file in one of the other PR's that I have open.
The checkstyle.xml file just the default google java style guideline.

**Pull request merged**